addon/test-waiter.ts

```diff
@@ -0,0 +1,97 @@
+import { IWaiter, WaiterName, ITestWaiterDebugInfo } from './types';
+import { register } from './waiter-manager';
+
+/**
+ * A class providing creation, registration and async waiting functionality.
+ *
+ * @public
+ * @class TestWaiter<T>
+ */
+export default class TestWaiter<T> implements IWaiter {
```

Comment:
Same as above, can you make an interface that both this and noop waiter implement (that is more specific than `IWaiter`)?

Reply:
Done.